Radix update

deptant opened the pull request:
This pull request updates the theme to use Radix. I have kept existing template, js and the Mega footer region. As this is a sub theme of Radix there is a dependency on the Radix theme being installed also. 

Do not install node modules from within the container. The current LTS version of node workd fine 'v14.15.4' Full instructions for building the css from SASS are available in the read me. `npm run watch` from outside the container will fire a browsersync window which can be useful. Further commands can be found in the webpack.js file.

mlsamuelson reviewed (approved):
Approved. Looks good. The theme looks like a nice, structured approach.

Before we merge, do we need to add yarn/npm calls into our build process for Webspark? I noticed pieces related to that in here, but I wasn't clear if the pieces needed have already been built out and included.

I'll include Radix in ASUWebPlatforms/webspark-upstream so when this hits, that dependency is in place.

mlsamuelson commented:
I've added the Radix dependency to the webspark-upstream codebase.

deptant commented:
Hi Michael,

In answer to the question about the build process, we will have to add npm
install and npm run production. Available scripts can be found in the
scripts section of the package.json file in case you prefer to have say a
dev version on the dev server and a prod version on staging.

It should be noted that the current LTS version is required for node
"14.16.0".

I've added a little bit more into this to demo field support for entity
blocks ready for Tuesday. Something that is not apparent from just looking
at the code is the fact that if you add a Drupal behaviour within a
component folder, then it is accessible via libraries. You can add an
include into the template for the library and it will only load when the
component is specifically loaded on a page, taking advantage of http2
multiple requests.



On Fri, Mar 5, 2021 at 7:48 PM Michael Samuelson <notifications@github.com>
wrote:

> I've added the Radix dependency to the webspark-upstream codebase.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ASUWebPlatforms/webspark-theme-renovation/pull/1#issuecomment-791642159>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AS3Q75OXE3KKQZCKGYG3NDTTCEYQ5ANCNFSM4YTMGUSQ>
> .
>


-- 
Anthony JW Averill
Devetree Developer

mlsamuelson commented:
@deptant Sounds good.
I'm okay with this being merged once we have the npm build bits figured out and merged for https://github.com/ASUWebPlatforms/webspark-upstream

I see two options here (maybe there's more?)
*We could add the npm install  and run production to the build process via [Composer Script](https://pantheon.io/blog/writing-composer-scripts) and possibly do something like https://stackoverflow.com/questions/25122187/using-composer-to-install-npm-and-bower-packages-on-production-i-e-no-devdepen . That said, Composer is a dependency manager, not a build tool, but in this limited capacity I don't find it too troublesome as a concept, as long as this is a workable idea.  
* Another option would be to pre-build and include the production bundle with the theme.

Which do you think makes the most sense?

deptant commented:
I'm going to have a good think about this today. It seems odd to have to
use Composer to build. Not seen that done before. Then again I've just got
used to the niceties of pipelines from Bitbucket and Acquia which makes
build processes far simpler. I'll get back to you later in your timezone.

On Mon, Mar 8, 2021 at 8:48 PM Michael Samuelson <notifications@github.com>
wrote:

> @deptant <https://github.com/deptant> Sounds good.
> I'm okay with this being merged once we have the npm build bits figured
> out and merged for https://github.com/ASUWebPlatforms/webspark-upstream
>
> I see two options here (maybe there's more?)
> *We could add the npm install and run production to the build process via Composer
> Script <https://pantheon.io/blog/writing-composer-scripts> and possibly
> do something like
> https://stackoverflow.com/questions/25122187/using-composer-to-install-npm-and-bower-packages-on-production-i-e-no-devdepen
> . That said, Composer is a dependency manager, not a build tool, but in
> this limited capacity I don't find it too troublesome as a concept, as long
> as this is a workable idea.
>
>    - Another option would be to pre-build and include the production
>    bundle with the theme.
>
> Which do you think makes the most sense?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ASUWebPlatforms/webspark-theme-renovation/pull/1#issuecomment-793066084>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AS3Q75M2Y7SW3XJFKDGQ2DTTCUZ3PANCNFSM4YTMGUSQ>
> .
>


-- 
Anthony JW Averill
Devetree Developer

deptant commented:
I have a better understanding about what we have now. For now the composer
solution seems the best option to me. At some point we should have a CI
process which does all this for us and runs tests and deploys to
testserver. I'll take a look at getting that running when I have some spare
time.

On Tue, Mar 9, 2021 at 8:49 AM Anthony Averill <
anthony.averill@deptagency.com> wrote:

> I'm going to have a good think about this today. It seems odd to have to
> use Composer to build. Not seen that done before. Then again I've just got
> used to the niceties of pipelines from Bitbucket and Acquia which makes
> build processes far simpler. I'll get back to you later in your timezone.
>
> On Mon, Mar 8, 2021 at 8:48 PM Michael Samuelson <notifications@github.com>
> wrote:
>
>> @deptant <https://github.com/deptant> Sounds good.
>> I'm okay with this being merged once we have the npm build bits figured
>> out and merged for https://github.com/ASUWebPlatforms/webspark-upstream
>>
>> I see two options here (maybe there's more?)
>> *We could add the npm install and run production to the build process via Composer
>> Script <https://pantheon.io/blog/writing-composer-scripts> and possibly
>> do something like
>> https://stackoverflow.com/questions/25122187/using-composer-to-install-npm-and-bower-packages-on-production-i-e-no-devdepen
>> . That said, Composer is a dependency manager, not a build tool, but in
>> this limited capacity I don't find it too troublesome as a concept, as long
>> as this is a workable idea.
>>
>>    - Another option would be to pre-build and include the production
>>    bundle with the theme.
>>
>> Which do you think makes the most sense?
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/ASUWebPlatforms/webspark-theme-renovation/pull/1#issuecomment-793066084>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AS3Q75M2Y7SW3XJFKDGQ2DTTCUZ3PANCNFSM4YTMGUSQ>
>> .
>>
>
>
> --
> Anthony JW Averill
> Devetree Developer
>


-- 
Anthony JW Averill
Devetree Developer
